fido2/hid/freebsd.py

```diff
@@ -28,20 +30,105 @@
 logger = logging.getLogger(__name__)
 
 
+devdir = "/dev/"
+
+device_re = re.compile("^uhid([0-9]+)$")
+vendor_re = re.compile("vendor=(0x[0-9a-fA-F]+)")
+product_re = re.compile("product=(0x[0-9a-fA-F]+)")
+sernum_re = re.compile('sernum="([^"]+)')
+
+libc = ctypes.CDLL(find_library("c"))
+
+USB_GET_REPORT_DESC = 0xC0205515
+
+
+class usb_gen_descriptor(ctypes.Structure):
+    _fields_ = [
+        (
+            "ugd_data",
+            ctypes.c_void_p,
+        ),  # TODO: check what COMPAT_32BIT in C header means
+        ("ugd_lang_id", ctypes.c_uint16),
+        ("ugd_maxlen", ctypes.c_uint16),
+        ("ugd_actlen", ctypes.c_uint16),
+        ("ugd_offset", ctypes.c_uint16),
+        ("ugd_config_index", ctypes.c_uint8),
+        ("ugd_string_index", ctypes.c_uint8),
+        ("ugd_iface_index", ctypes.c_uint8),
+        ("ugd_altif_index", ctypes.c_uint8),
+        ("ugd_endpt_index", ctypes.c_uint8),
+        ("ugd_report_type", ctypes.c_uint8),
+        ("reserved", ctypes.c_uint8 * 8),
+    ]
+
+
 def open_connection(descriptor):
     return FileCtapHidConnection(descriptor)
 
 
+def _get_report_data(fd, report_type):
+    data = ctypes.create_string_buffer(b"\000" * 4096)
```

Comment:
I think it's a bit cleaner to just pass the size to `create_string_buffer` since that will be zeroed out anyway. Do note that `create_string_buffer(4096)` will create a buffer of size 4096, whereas create_string_buffer(b"\000" * 4096) will add a NULL byte at the end, resulting in a a buffer of size 4097.

Reply:
Sure! That's much nicer. I just wasn't sure about the zero initialization.